packages/orchestration/src/orchestration-types.ts

```diff
@@ -56,3 +58,42 @@ export interface OrchestrationModuleConfig {
    */
   grounding?: GroundingModuleConfig;
 }
+
+/**
+ * Represents a filter configuration for the Document Grounding Service.
+ *
+ */
+export type DocumentGroundingServiceFilter = Pick<
```

Comment:
[q] Can't this be achieved with a type union or extension? This way we'll have to remember to add new fields..

Reply:
I tried. The below doesn't work (optional modifier is not applied :

```
export type DocumentGroundingServiceFilter = DocumentGroundingFilter & {
      data_repository_type?: DataRepositoryType;
     }
```

I also tried:

```
export type DocumentGroundingServiceFilter = Omit<DocumentGroundingFilter, 'data_repository_type'> & {
      data_repository_type?: DataRepositoryType;
     }
```

This causes issues with auto-completion as `DocumentGroundingFilter` type in extensible (has Record<string, any>). 

But I agree to your point about keeping track of properties. I'll add a test to ensure we are always consistent.

Reply:
With the latest update (additionalProperties: false in GroundingModuleConfig), a simpler type is possible 👍🏽